src/wp-includes/user.php

```diff
@@ -3629,53 +3629,101 @@ function _wp_get_current_user() {
 function send_confirmation_on_profile_email() {
 	global $errors;
 
-	$current_user = wp_get_current_user();
 	if ( ! is_object( $errors ) ) {
 		$errors = new WP_Error();
 	}
 
-	if ( $current_user->ID != $_POST['user_id'] ) {
-		return false;
-	}
+	$user       = get_user_by( 'id', $_POST['user_id'] );
+	$email_sent = send_user_email_change_confirmation_email( $user, $_POST['email'] );
 
-	if ( $current_user->user_email != $_POST['email'] ) {
-		if ( ! is_email( $_POST['email'] ) ) {
+	if ( is_wp_error( $email_sent ) ) {
+		// WP_Error::copy_errors() with the addition of adding data.
+		foreach ( $email_sent->get_error_codes() as $code ) {
 			$errors->add(
-				'user_email',
-				__( '<strong>Error:</strong> The email address is not correct.' ),
+				$code,
+				$email_sent->get_error_message( $code ),
 				array(
 					'form-field' => 'email',
 				)
 			);
-
-			return;
 		}
+	}
 
-		if ( email_exists( $_POST['email'] ) ) {
-			$errors->add(
-				'user_email',
-				__( '<strong>Error:</strong> The email address is already used.' ),
-				array(
-					'form-field' => 'email',
-				)
-			);
-			delete_user_meta( $current_user->ID, '_new_email' );
+	if ( true === $email_sent ) {
+		$_POST['email'] = $user->user_email;
+	}
+}
 
-			return;
-		}
+/**
+ * Send the 'confirm your email' email.
+ *
+ * @since x.x
+ *
+ * @param WP_User $user  The user to act upon.
+ * @param string  $email The new email address.
+ * @return null|true|WP_Error true if email sent, WP_Error on error, and null otherwise.
+ */
+function send_user_email_change_confirmation_email( $user, $email ) {
+	if ( $user->user_email == $email ) {
+		return;
+	}
 
-		$hash           = md5( $_POST['email'] . time() . wp_rand() );
-		$new_user_email = array(
-			'hash'     => $hash,
-			'newemail' => $_POST['email'],
+	if ( ! is_email( $email ) ) {
+		return new WP_Error(
+			'user_email',
+			__( '<strong>Error:</strong> The email address is not correct.' )
+		);
+	}
+
+	if ( email_exists( $email ) ) {
+		delete_user_meta( $user->ID, '_new_email' );
```

Comment:
I'm not sure why this error message is present within this function, it seems like it should be in `edit_user()` rather than on the email-change-email function.